### ISSUE TITLE
futures::Future trait implementation not available for IAsyncOperation<T> in v0.58 for some APIs

### ISSUE THREAD

**Issue by qdot** [microsoft/windows-rs#3197]
### Summary

Trying to update to 0.58 in https://github.com/deviceplug/btleplug, now getting errors on all async calls to the WinRT Bluetooth APIs:

```
error[E0277]: `IAsyncOperation<GattCommunicationStatus>` is not a future
  --> src\winrtble\ble\characteristic.rs:72:32
   |
72 |         let result = operation.await?;
   |                               -^^^^^
   |                               ||
   |                               |`IAsyncOperation<GattCommunicationStatus>` is not a future
   |                               help: remove the `.await`
   |
   = help: the trait `futures::Future` is not implemented for `IAsyncOperation<GattCommunicationStatus>`, which is required by `IAsyncOperation<GattCommunicationStatus>: std::future::IntoFuture`
   = note: IAsyncOperation<GattCommunicationStatus> must be a future or must implement `IntoFuture` to be awaited
   = note: required for `IAsyncOperation<GattCommunicationStatus>` to implement `std::future::IntoFuture
```

I looked through the CHANGELOG but it's rather dense for this version update and I'm not sure what I'm missing.

Our last library version was released using v0.52, and updating to v0.57 still seems to work, it's just moving to v0.58 that seems to be causing this issue.

**Comment by kennykerr** [comment]
This was removed in #3142 out of safety concerns, but there is an attempt to bring it back in #3177.

**Comment by kennykerr** [comment]
Closing in favor of #342

**Comment by makepaddev** [comment]
I have the same problem, is this fixed now somehow? I can't seem to find a clear answer in all the related links. How do i get to call await?

**Comment by kennykerr** [comment]
Yes, this has been fixed as of #3213 - there just hasn't been a release of the `windows` crate with this update.

**Comment by makepaddev** [comment]
Ok. I've reverted to 0.56 in the meantime, also in 0.57/0.58 the 'implement' COM macro complains about core any not being available

**Comment by fenhl** [comment]
Is this fix still not released? Based on the date #3213 was merged, it seems like it should be, but I'm trying to upgrade my `windows` dependency from 0.57 to 0.61 and am running into an issue with this code:

```rust
LockScreen::SetImageStreamAsync(&FileRandomAccessStream::OpenAsync(&HSTRING::from(&*path), FileAccessMode::Read)?.await?)?;
```

The compiler says “`windows_future::bindings::IAsyncOperation<IRandomAccessStream>` is not a future”.

**Comment by fenhl** [comment]
Figured it out: I needed to add an explicit dependency on the `windows-future` crate to enable its `std` feature, since the `IntoFuture` impl is gated behind that feature, and `windows`'s `std` feature does not enable it.
